addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -2176,4 +2181,32 @@ MockServer.include({
             starred_counter: this._getRecords('mail.message', [['starred_partner_ids', 'in', user.partner_id]]).length,
         };
     },
+    /**
+     * Simulate the `notify_cancel_by_type` on `mail.thread` .
+     * Note that this method is overridden by snailmail module but not simulated here.
+     */
+    _mockMailThreadNotifyCancelByType(model, notificationType) {
+        const self = this;
+        const authorId = this.currentPartnerId;
+
+        let notificationMails = this._getRecords('mail.notification', [
+            ['notification_type', '=', notificationType],
+            ['notification_status', 'in', ['bounce', 'exception']],
+        ]).map(notification => ({
+            notification: notification,
+            mail: self._getRecords('mail.message', [['id', '=', notification.mail_message_id]])[0]
+        })).filter(notifMail => notifMail.mail.model === model && notifMail.mail.author_id === authorId);
+
+        notificationMails.forEach(notifMail => notifMail.notification.notification_status = 'canceled');
+        const mailIds = notificationMails.map(notifMail => notifMail.mail.id);
+
+        // Following lines mock mail.message._notify_message_notification_update on the selected mail
+        this._widget.call('bus_service', 'trigger', 'notification', [{
+            type: 'mail.message/notification_update',
+            payload: {
+                elements: this._mockMailMessage_MessageNotificationFormat(mailIds),
+            },
+        }]);
+        return true;
```

Comment:
Looks like a useless return value (also useless in python)

Reply:
I have removed it. Just to let you know why: I have added that return value because of the tutorial I have followed the first week. It is written there "Finally, a public method should always return something so that it can be called through XML-RPC. When in doubt, just return True" (https://www.odoo.com/documentation/15.0/developer/howtos/rdtraining/10_actions.html)

Reply:
I see, I wasn't aware of that recommendation. Maybe a relic of the past? Or maybe it was meant specifically for web client action manager, and not JSON RPC in general?

If you removed it and it still works, the tutorial should be updated. (and I say `if` to be careful, but we already have plenty of other examples where nothing is returned and it works just fine)